src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -238,6 +247,88 @@ async fn compact_shared_buffer(
     }
 }
 
+/// Merge multiple batches into a larger one
+pub async fn merge_imms_in_memory(
+    table_id: TableId,
+    instance_id: LocalInstanceId,
+    imms: Vec<ImmutableMemtable>,
+    memory_tracker: Option<MemoryTracker>,
+) -> HummockResult<ImmutableMemtable> {
+    let mut range_tombstone_list = Vec::new();
+    let mut kv_count = 0;
+    let mut epochs = vec![];
+    let mut merged_size = 0;
+    let mut merged_imm_ids = Vec::with_capacity(imms.len());
+
+    let mut imm_iters = Vec::with_capacity(imms.len());
+    for imm in imms {
+        assert!(imm.kv_count() > 0, "imm should not be empty");
+        assert_eq!(
+            table_id,
+            imm.table_id(),
+            "should only merge data belonging to the same table"
+        );
+
+        merged_imm_ids.push(imm.batch_id());
+        epochs.push(imm.min_epoch());
+        kv_count += imm.kv_count();
+        merged_size += imm.size();
+        range_tombstone_list.extend(imm.get_delete_range_tombstones());
+        imm_iters.push(imm.into_forward_iter());
+    }
+    range_tombstone_list.sort();
+    epochs.sort();
+
+    // use merge iterator to merge input imms
+    let mut mi = UnorderedMergeIteratorInner::new(imm_iters);
```

Comment:
We shall use `OrderedMergeIteratorInner` in case that we may support write-anytime in the future.